advisories/a63162.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Preliminary Technical Advisory 63162
+summary: Invalid incremental backups under certain circumstances
+toc: true
+---
+
+Preliminary date: April 6, 2021
+
+**Preliminary, and subject to change**
+
+## Description
+
+Cockroach Labs has discovered a bug relating to incremental backups, which under certain circumstances may result in data loss when restoring from an incremental backup. This is a **preliminary notice** in the interest of transparency, and will change.
+
+If an incremental backup coincides with an in-progress index creation or rebuild, `RESTORE`, or `IMPORT`, it is possible that the incremental backup will not include all of the indexed, restored or imported data.
```

Comment:
technically if _any_ backup -- full or incremental -- captures an index in-process of being built *and then* an incremental backup captures it again but as complete, that's the issue. So while you do need an incremental, just by definition to have passage of time and thus to have an index that was backfilling and then isn't, it doesn't need to be the incremental that is run while the index is building -- it is the incremental after the index has been built that is the problem.

So e.g. this sequence can't trust the restore to 315 inc backup either:
2:55: Create index
3:00: Full Backup
3:05: index create finished 
3:15: incremental backup

That's because it only backs up to the _previous_ backup (full or inc) but the index backfill which ran _since then_ wrote data before that backup, which it won't capture.

Reply:
OK, how about: 

If a backup coincides with an in-progress index creation or rebuild, `RESTORE`, or `IMPORT`, it is possible that a subsequent incremental backup will not include all of the indexed, restored or imported data.

Reply:
That seems accurate, could add optionally specify that it is the subsequent incremental backup _after the operation completes_ but may not matter too much

Reply:
OK, I made a change, just pushed:

If a backup coincides with an in-progress index creation (backfill), `RESTORE`, or `IMPORT`, it is possible that a subsequent incremental backup will not include all of the indexed, restored or imported data.

(I don’t want to make claims about rarity or likelihood here. Users can draw those conclusions based on the facts we lay out.)

Reply:
(Whoops, I think I repeated a comment above.)

I think ‘after the operation completes’ doesn’t add to the understanding, even though it’s accurate.